sdk/transport_policies/curl/src/az_curl.c

```diff
@@ -291,9 +292,10 @@ static int32_t _az_http_client_curl_upload_read_callback(
 
   int32_t data_length = az_span_length(*mut_span);
   int32_t to_copy = (data_length < curl_size) ? data_length : curl_size;
-  AZ_RETURN_IF_FAILED(
```

Comment:
I think we need a super great and amazing comment for this method to mention what it does.
It is being called as many times as it requires to upload something right?
Every time it gets called, we would upload a chunk of data?  that chuck needs to be copied to ptr with max = data_lenght?
And user_data is pointing to the start of what we need to upload?
If so, every time this is called, we need to move the userdata pointer += curl_size ?  so the next time we add the next chunck?  or is it something else?

Reply:
More less that is what its doing.  I've added a comment to that sense with references back to the curl documentation.